filters/src/passthrough.cpp

```diff
@@ -273,6 +273,106 @@ pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
   removed_indices_->resize (nr_removed_p);
 }
 
+void
+pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (std::vector<int> &indices)
+{
+  // The arrays to be used
+  indices.resize (indices_->size ());
+  removed_indices_->resize (indices_->size ());
+  int oii = 0, rii = 0;  // oii = output indices iterator, rii = removed indices iterator
+  const std::uint32_t xyz_offset[3] = {input_->fields[x_idx_].offset, input_->fields[y_idx_].offset, input_->fields[z_idx_].offset};
+  PCL_DEBUG ("[pcl::%s<pcl::PCLPointCloud2>::applyFilter] Field offsets: x: %u, y: %u, z: %u.\n", filter_name_.c_str (), xyz_offset[0], xyz_offset[1], xyz_offset[2]);
+
+  // Has a field name been specified?
+  if (filter_field_name_.empty ())
+  {
+    // Only filter for non-finite entries then
+    for (const auto ii : indices)  // ii = input index
+    {
+      float pt[3];
+      memcpy (&pt[0], &input_->data[ii * input_->point_step + xyz_offset[0]], sizeof(float));
+      memcpy (&pt[1], &input_->data[ii * input_->point_step + xyz_offset[1]], sizeof(float));
+      memcpy (&pt[2], &input_->data[ii * input_->point_step + xyz_offset[2]], sizeof(float));
+      // Non-finite entries are always passed to removed indices
+      if (!std::isfinite (pt[0]) ||
+          !std::isfinite (pt[1]) ||
+          !std::isfinite (pt[2]))
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
+      indices[oii++] = ii;
+    }
+  }
+  else
+  {
+    // Attempt to get the field name's index
+    int distance_idx = pcl::getFieldIndex (*input_, filter_field_name_);
+    if (distance_idx == -1)
+    {
+      PCL_WARN ("[pcl::%s::applyFilter] Unable to find field name in point type.\n", getClassName ().c_str ());
+      indices.clear ();
+      removed_indices_->clear ();
+      return;
+    }
+
```

Comment:
There's not check here if the fields type to be filtered is a 32bit float as it exists in the other overload.

Reply:
Ok, so a check if it is a float/double/something else like with x, y, z